README
Titta på example.php

En stark rekomendation är att generera ett fast auth-nyckel.
Gnerera nyckel med getAuthorizationKey() och lägg in nycklen när klassen instansieras lik koden nedan:

new SwedbankJson($username, $password, 'Min-fasta-auth-nyckel');

CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

## [Unreleased][unreleased]

## [1.0.0] - 2021-08-29
### Added
- New authentication method:  Mobile BankID on same device.
- New authentication method:  Mobile BankID QR code. Pull request [#43](https://github.com/walle89/SwedbankJson/pull/43). Thank you [@matthiaspalmer](https://github.com/matthiaspalmer) and [@NisseDILLIGAF](https://github.com/NisseDILLIGAF) for your contributions.
- Automatic updates of or only remotely fetch app data metadata. Issue [#40](https://github.com/walle89/SwedbankJson/issues/40).
- Fetch transaction details from transaction row. Issue [#38](https://github.com/walle89/SwedbankJson/issues/38).
- Method transferBaseInfo().
- Method transferRegisterPayment().
- Method transferListRegistered().
- Method transferListConfirmed().
- Method transferConfirmPayments().

### Changed
- All auth classes now requires `AppData` object.
- Moved from version 4 to version 5 of the Swedbank API.
- setBaseUri() method now need API version number included.

### Removed
- Mobile BankID with Personal identity number
- Bank types "swedbank_ung" and "sparbanken_ung".
- Warning for the "swedbank_företag" bank type.
- To set BankType string in auth classes.
- Method baseInfo(), replaced with transferBaseInfo().
- Method registerTransfer(), replaced with transferRegisterPayment().
- Method listRegisteredTransfers(), replaced with transferListRegistered().
- Method listConfirmedTransfers(), replaced with transferListConfirmed().
- Method deleteTransfer(), replaced with transferDeletePayment().
- Method confirmTransfers(), replaced with transferConfirmPayments().

## [0.7.6] - 2020-01-23
### Fixed
- Updated app version for all Swedbank apps in order to support the new minimal required version to use the API.

## [0.7.5] - 2019-12-09
### Fixed
- Updated app version for all Swedbank apps in order to support the new minimal required version to use the API. Issue [#39](https://github.com/walle89/SwedbankJson/issues/39).

## [0.7.4] - 2019-10-24
### Fixed
- Updated app version for all Swedbank apps (expect youth apps) in order to support the new minimal required version to use the API. Issue [#39](https://github.com/walle89/SwedbankJson/issues/39).

## [0.7.3] - 2019-10-07
### Fixed
- Updated app version for all Swedbank apps in order to support the new minimal required version to use the API. Issue [#39](https://github.com/walle89/SwedbankJson/issues/39).

## [0.7.2] - 2019-05-26
### Fixed
- Updated app version for all Swedbank apps in order to support the new minimal required version to use the API.
- Stop endless looping when logout API call fails. Resulting in a Fatal error: Allowed memory message.

## [0.7.1] - 2016-10-22
### Changed
- Updated app version for all Swedbank apps.
- Updated UUID dependency.

## [0.7.0] - 2016-07-25
### Security
- Guzzle set to minimum version of 6.2.1 because of HTTP_PROXY vulnerability.

### Added
- Transfer money between accounts.
- Option to log HTTP requests to file for debugging purposes.

### Changed
- Translated the documentation to English.
- Improved documentation. Now with API response samples.
- Updated app version for all Swedbank apps.
- Refactor getchallenge() to getChallenge(). Method names are case-insensitive, should have no effect.
- Refactor setchallengeResponse() to setChallengeResponse(). Method names are case-insensitive, should have no effect.

### Deprecated
- Refactor of confirmTransfer() to confirmTransfers().

## [0.6.1] - 2016-02-07
### Added
- Documentation for quick balance.

### Fixed
- Cleanup() didn't clean up saved session.
- UserException namespace for Appdata.
- Documentation links

## [0.6.0] - 2016-01-31
### Added
- New authentication method: Mobile BankID.
- New authentication method: No login.

### Changed
- Upgraded to Guzzle 6, therefore PHP 5.5 or later is required.
- Updated app version for all Swedbank apps.
- Improved error handling.
- Improved documentation.

### Removed
- Automatic sign out - You have to manually use terminate() to sign out.

## [0.5.2] - 2015-04-21
### Changed
- Update of Guzzle dependencies.

### Fixed
- Getchallenge() didn't save the challenge within a session. Issue [#22](https://github.com/walle89/SwedbankJson/issues/22).

## [0.5.1] - 2015-03-09
### Changed
- Update of Guzzle.

## [0.5.0] - 2015-03-09
### Added
- Support for authentication method Security token with one time code (see [README.md]).

### Changed
- The SwedbankJson object have been redesigned to support security token login. Read more about it in [README.md].
- New way to set AppID. It's backward compatible with the old method.
- Updated app version for all Swedbank apps.
- Updated documentation with guide for Windows users.

## [0.4.0] - 2015-02-13
### Added
- Quick balance - Show, activate and deactivate Quick balance.
- Reminders - Get the number of rejected payments, payments signed, unsigned transfers and incoming e-invoices.
- Baseinfo - Account list grouped by type.

### Changed
- AccountDetails does not longer require AccountID. If no AccountID is provided, it will fallback to default account.
- Setting ProfileID is no longer required. If ProfileID is not provided, it will fallback to default profile. 
- Updated app version for all Swedbank apps.
- Updated documentation.
- Updated dependencies of UDID and Guzzle.

## [0.3.4] - 2014-06-18
### Added
- Support for Sparbanken Företag.

### Changed
- Updated app version for Swedbank and Sparbanken.
- Updated documentation.

### Deprecated
- Bank type "swedbank_företag" have changed to "swedbank_foretag".

## [0.3.3] - 2014-05-28
### Added
- MIT License file.

### Changed
- Improved documentation.
- Updated to Guzzle 4.1.

### Fixed
- Removing cookie file after each run. 

## [0.3.2] - 2014-05-15
### Fixed
- Issue [#8](https://github.com/walle89/SwedbankJson/issues/8). Linux uses a case sensitive file system, unlike OS X.

## [0.3.1] - 2014-05-15
### Changed
- Changed namespace.

src/appdata.php
<?php
/**
 * Inställningar öfr
 *
 * @package Projectnamn
 * @author walle
 * Date: 2014-05-13
 * Time: 21:40
 */
namespace walle89\SwedbankJson;

/**
 * Class appdata
 */
class AppData
{
    private static $appData = array(
        'swedbank'          => array('appID' => 'fjUHYbQjupOQspeG', 'useragent' => 'SwedbankMOBPrivateIOS/3.7.1_(iOS;_7.1.1)_Apple/iPhone5,2'),
        'sparbanken'        => array('appID' => 'WXessEsk4s1ZE4GO', 'useragent' => 'SavingbankMOBPrivateIOS/3.7.1_(iOS;_7.1.1)_Apple/iPhone5,2'),
        'swedbank_ung'      => array('appID' => 'XlqIpRxVTTgHuFnj', 'useragent' => 'SwedbankMOBYouthIOS/1.4.0_(iOS;_7.1.1)_Apple/iPhone5,2'),
        'sparbanken_ung'    => array('appID' => 'XOn0HeGwPmELslnO', 'useragent' => 'SavingbankMOBYouthIOS/1.4.0_(iOS;_7.1.1)_Apple/iPhone5,2'),
        'swedbank_företag'  => array('appID' => '1e7ZaX5kq3Dk4mCK', 'useragent' => 'SwedbankMOBCorporateIOS/1.4.1_(iOS;_7.1.1)_Apple/iPhone5,2'),
    );

    public static function bankAppId($bankID)
    {
        return self::$appData[$bankID];
    }
}